### PR TITLE
chore(deps): update some dependencies

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/KIT-282

did it myself to replace https://github.com/coveo/ui-kit/pull/3795. renovate doesnt work well when the pr stays up. The dependency table is desynced and I kept getting a bunch of errors.

### CODE REVIEW

**Review by louis-bompart — COMMENTED** [review]
lgtm overall, good idea. However let's tread carefuly around some of them,

File: packages/headless/package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "fast-equals": "5.0.1",
     "navigator.sendbeacon": "0.0.20",
     "node-abort-controller": "^3.0.0",
-    "pino": "8.21.0",
```
[the remark this answers is not in the document]
Comment (alexprudhomme) [reply packages/headless/package.json:64]: The only breaking change in that major version is  :

> ⚠ BREAKING CHANGES
Dropped support for Node.js 14, 16, & 19

should be good !

File: utils/release/package.json [hunk 2]
```diff
@@ -7,11 +7,11 @@
   "dependencies": {
     "@coveo/semantic-monorepo-tools": "2.4.31",
     "@npmcli/arborist": "7.5.1",
-    "@octokit/auth-app": "6.1.1",
+    "@octokit/auth-app": "7.1.0",
     "async-retry": "1.3.3",
-    "conventional-changelog-angular": "7.0.0",
+    "conventional-changelog-angular": "8.0.0",
     "dependency-graph": "1.0.0",
-    "octokit": "3.2.1",
```
[the remark this answers is not in the document]
Comment (alexprudhomme) [reply utils/release/package.json:14]: @octokit/auth-app : should be good. we dont use the `oauth` type

> BREAKING CHANGES
package is now ESM
remove type "oauth" that was previously deprecated

conventional-changelog-angular: the changes are long but we only use a singular function in our code `angularChangelogConvention();` the major doesnt affect us.

https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/CHANGELOG.md#800-2024-04-26

File: packages/headless/package.json [hunk 3]
```diff
@@ -72,7 +72,7 @@
     "@microsoft/api-extractor-model": "7.28.16",
     "@microsoft/tsdoc": "0.14.2",
     "eslint-plugin-canonical": "4.18.0",
-    "execa": "8.0.1",
```
[the remark this answers is not in the document]
Comment (alexprudhomme) [reply packages/headless/package.json:75]: thanks for this comment. I was able to remove the execa dependency from the root project (it was used in a old unused storybook script  in `scripts/bin/install-isolated.mjs`)

For the dependency in headless, it is used here below in a documentation script. The changes in the major are long but they are all related to complex options in the execa function that we dont use. (https://github.com/sindresorhus/execa/releases/tag/v9.0.0)

https://github.com/coveo/ui-kit/blob/KIT-282-deps/packages/headless/scripts/extract-documentation.mjs

File: packages/headless/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (olamothe) [reply packages/headless/package.json:64]: We will ourselves drop support of those NodeJS version only in v3. So I think its too early to update this one.

[the remark this answers is not in the document]
Comment (alexprudhomme) [reply packages/headless/package.json]: I reverted pino

**Review by olamothe — APPROVED** [review]

**Comment by alexprudhomme** [comment]
im done fighting this is a waste of time. bunch of problems

**Pull request closed without merging**